patches/minecraft/net/minecraft/entity/item/BoatEntity.java.patch

```diff
@@ -1,6 +1,29 @@
 --- a/net/minecraft/entity/item/BoatEntity.java
 +++ b/net/minecraft/entity/item/BoatEntity.java
-@@ -459,7 +459,7 @@
+@@ -178,21 +178,7 @@
+    }
+ 
+    public Item func_184455_j() {
+-      switch(this.func_184453_r()) {
+-      case OAK:
+-      default:
+-         return Items.field_151124_az;
+-      case SPRUCE:
+-         return Items.field_185150_aH;
+-      case BIRCH:
+-         return Items.field_185151_aI;
+-      case JUNGLE:
+-         return Items.field_185152_aJ;
+-      case ACACIA:
+-         return Items.field_185153_aK;
+-      case DARK_OAK:
+-         return Items.field_185154_aL;
+-      }
++      return this.func_184453_r().getBoatItem();
```

Comment:
This patch can be so much smaller by moving this ABOVE the switch and letting vanilla types use the switch.